sky/skypilot_config.py

```diff
@@ -151,6 +152,9 @@ def __init__(self,
 
 SKYPILOT_CONFIG_LOCK_PATH = '~/.sky/locks/.skypilot_config.lock'
 
+_WARNED_DISALLOWED_KEYS_CACHE: typing.Set[Tuple[str, Tuple[str, ...]]] = set()
+_WARNED_DISALLOWED_KEYS_CACHE_LOCK = threading.Lock()
+
```

Comment:
```suggestion
_WARNED_DISALLOWED_KEYS_CACHE_SIZE = 1000
```
so that we're not defining the value inline